flink-connectors/flink-connector-aws-kinesis-data-streams/src/main/java/org/apache/flink/connector/kinesis/sink/KinesisDataStreamsException.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.kinesis.sink;
+
+/**
+ * A {@link RuntimeException} wrapper indicating the exception was thrown from the Kinesis Data
+ * Streams Sink.
+ */
+class KinesisDataStreamsException extends RuntimeException {
+
+    public KinesisDataStreamsException(final String message) {
+        super(message);
+    }
+
+    public KinesisDataStreamsException(final String message, final Throwable cause) {
+        super(message, cause);
+    }
+
+    /**
+     * When the flag {@code failOnError} is set in {@link KinesisDataStreamsSinkWriter}, this
+     * exception is raised as soon as any exception occurs when KDS is written to.
+     */
+    static class KinesisDataStreamsFailFastException extends KinesisDataStreamsException {
```

Comment:
nit: The name of this class is not sitting well with me. I think something more general like `KinesisDataStreamsRuntimeException` seems like a better fit. Thoughts?

Reply:
This exception is **only** used to fail fast to deny request retries (the class is not extendible for reuse in case there is another case of runtime exceptions in datastreams, i.e. the error message is fixed). I think the intended use of this class is to be specific while the `KinesisDataStreamsException` is general.